sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/interface.scala

```diff
@@ -655,21 +646,35 @@ case class CatalogTable(
     }
     if (comment.isDefined) map += "Comment" -> JString(comment.get)
     if (collation.isDefined) map += "Collation" -> JString(collation.get)
-    if (tableType == CatalogTableType.VIEW && viewText.isDefined) {
-      map += "View Text" -> JString(viewText.get)
-    }
-    if (tableType == CatalogTableType.VIEW && viewOriginalText.isDefined) {
-      map += "View Original Text" -> JString(viewOriginalText.get)
-    }
-    if (SQLConf.get.viewSchemaBindingEnabled && tableType == CatalogTableType.VIEW) {
-      map += "View Schema Mode" -> JString(viewSchemaMode.toString)
-    }
-    if (viewCatalogAndNamespace.nonEmpty && tableType == CatalogTableType.VIEW) {
-      import org.apache.spark.sql.connector.catalog.CatalogV2Implicits._
-      map += "View Catalog and Namespace" -> JString(viewCatalogAndNamespace.quoted)
-    }
-    if (viewQueryOutputColumns != JNull) {
-      map += "View Query Output Columns" -> viewQueryOutputColumns
+
+    if (tableType == CatalogTableType.VIEW) {
+      if (viewText.isDefined) {
+        map += "View Text" -> JString(viewText.get)
+      }
+      if (viewOriginalText.isDefined) {
+        map += "View Original Text" -> JString(viewOriginalText.get)
+      }
+      if (SQLConf.get.viewSchemaBindingEnabled) {
+        val viewSchemaModeInfo = Try(viewSchemaMode.toString).getOrElse("UNKNOWN")
+        map += "View Schema Mode" -> JString(viewSchemaModeInfo)
+      }
+      val viewCatalogAndNamespaceInfos = Try(viewCatalogAndNamespace).getOrElse(Seq.empty)
+      if (viewCatalogAndNamespaceInfos.nonEmpty) {
+        import org.apache.spark.sql.connector.catalog.CatalogV2Implicits._
+        map += "View Catalog and Namespace" -> JString(viewCatalogAndNamespaceInfos.quoted)
+      }
+      val viewQueryOutputColumns: JValue = Try {
```

Comment:
can we use standard try-catch and swallow NonFatal errors?

Reply:
hm, doesn't Try do that (catch only NonFatal, ie https://github.com/scala/scala/blob/v2.13.2/src/library/scala/util/Try.scala#L204)  not sure i miss something